analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StreamReadStatement.CSharp10.cs

```diff
@@ -9,11 +9,14 @@ public class StreamReadStatement
     {
         public StreamReadStatement(string fileName)
         {
-            using (var stream = File.Open(fileName, FileMode.Open))
-            {
-                var result = new byte[stream.Length];
-                (int a, int b) = (stream.Read(result, 0, (int)stream.Length), 42); // FN
-            }
+            using var stream = File.Open(fileName, FileMode.Open);
+            var result = new byte[stream.Length];
+            (int a, int b) = (stream.Read(result, 0, (int)stream.Length), 42); // Compliant The result is assigned to a
+            _ = stream.Read(result, 0, (int)stream.Length);                    // FN. The result is discarded
+            (_, var c) = (stream.Read(result, 0, (int)stream.Length), 42);     // FN. The result is discarded
+            _ = stream.Read(result, 0, (int)stream.Length) is { };             // FN. The result is discarded
+            _ = stream.Read(result, 0, (int)stream.Length) is { } _;           // FN. The result is discarded
+            _ = (stream.Read(result, 0, (int)stream.Length), 42) is (_, _);    // FN. The result is discarded
```

Comment:
These are some coding patterns, where the return value is read, but discarded. To support these and other related patterns, flow analysis is needed IMO. We should document it in the Kanban board.

Reply:
Good point! This would be an Improvement ticket

Reply:
@martin-strecker-sonarsource did you capture this somewhere?

Reply:
Not yet. There are three issues with this PR now:
* [ ] Discards not handled properly
* [ ] Upcoming ReadAtLeast.. methods not handled properly
* [ ] RemoveConditionalAccess needs some companioning methods from roslyn (e.g. [This one](https://github.com/dotnet/roslyn/blob/2aed89bfee5d5318cb62108be76365befb656c38/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs#L295-L371))

Where should we document these?

Reply:
For **Upcoming ReadAtLeast.. methods not handled properly** -> you can add a comment to the [C# 11 trello card](https://trello.com/c/4HMy7PvW/527-survey-impact-of-c11-features-on-our-rules-eg-stringsyntax)

For **RemoveConditionalAccess needs some companioning methods from roslyn (e.g. This one)** - this can be a standalone ticket

For `Discards not handled properly` - what does this mean? Does it generate FPs or FNs? This should be an "improvement" card or an FP card in our GitHub backlog.